pkg/diff/checkpoint.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"go.uber.org/zap"
+)
+
+var (
+	// for chunk: means this chunk's data is equal
+	// for table: means this all chunk in this table is equal(except ignore chunk)
+	successState = "success"
+
+	// for chunk: means this chunk's data is not equal
+	// for table: means some chunks' data is not equal or some chunk check failed in this table
+	failedState = "failed"
+
+	// for chunk: means meet error when check, don't know the chunk's data is equal or not equal
+	// for table: don't have this state
+	errorState = "error"
+
+	// for chunk: means this chunk is not in check
+	// for table: all the chunk in this table is not in check
+	notCheckedState = "not_checked"
+
+	// for chunk: means this chunk is checking
+	// for table: some chunks in this table is checking
+	checkingState = "checking"
+
+	// for chunk: this chunk is ignored. if sample is not 100%, will ignore some chunk
+	// for table: don't have this state
+	ignoreState = "ignore"
+)
```

Comment:
if we add some interface for checkpoint, it would fine for unit test, cuz we can mock it